extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js

```diff
@@ -155,29 +139,117 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.insertInfoboxTemplate = function () {
 };
 
 /**
- * Add the infobox template names to the dialog's SelectWidget
- *
+ * @desc creates infobox item option wiget
+ * @param {Object} data
+ * @returns {OO.ui.DecoratedOptionWidget}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createInfoboxItemOptionWidget = function ( data ) {
+	return new OO.ui.DecoratedOptionWidget( {
+		data: data.title,
+		label:  data.title
+	} );
+};
+
+/**
+ * @desc creates select widget with indobox option widgets
+ * @param {OO.ui.DecoratedOptionWidget[]} items
+ * @returns {OO.ui.OoUiSelectWidget}
+ */
+ve.ui.WikiaInfoboxInsertDialog.prototype.createInfoboxSelectWidget = function ( items ) {
+	var select = new OO.ui.SelectWidget();
+
+	select.addItems( items );
+
+	return select;
+};
+
+/**
+ * @desc creates dialog content
  * @param {Object} data Response data from API
+ * @returns {Promise}
  */
-ve.ui.WikiaInfoboxInsertDialog.prototype.showResults = function ( data ) {
-	var i,
-		items = [],
-		infoboxes = data.query ? data.query.allinfoboxes : [];
-
-	if ( infoboxes.length > 0 ) {
-		for ( i = 0; i < infoboxes.length; i++ ) {
-			items.push(
-				new OO.ui.DecoratedOptionWidget({
-					data: infoboxes[i].title,
-					label:  infoboxes[i].title
-				})
-			);
-		}
-		this.select.addItems( items );
+ve.ui.WikiaInfoboxInsertDialog.prototype.createDialogContent = function ( data ) {
+	var deferred = $.Deferred(),
+		infoboxes = data.query ? data.query.allinfoboxes : [],
+		items = infoboxes.map( this.createInfoboxItemOptionWidget );
```

Comment:
maybe we can name this two vars in more descriptive way, e.g. `infoboxTemplates` and `infoboxTemplateOptions` ? Just a suggestion.